pkg/sql/sqlbase/roundtrip_format_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqlbase
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// TestParseDatumStringAs tests that datums are roundtrippable between
+// printing with FmtExport and ParseDatumStringAs, but with random datums.
+// This test lives in sqlbase to avoid dependency cycles when trying to move
+// RandDatumWithNullChance into tree.
+func TestRandParseDatumStringAs(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	tests := []*types.T{
```

Comment:
This should use types.Scalar and add on any special things:

```
tests := append([]*types.T{
types.MakeTimestamp(0),
types.MakeTimestamp(3),
types.MakeTimestamp(6),
...,
}, types.Scalar...)
```

this way when new types are added to that list they will be tested here.

Reply:
Ok, will do.